### ISSUE TITLE
Process route information as a mediator

### ISSUE THREAD

**Issue by fredo** [raiden-network/raiden#6849]
**Goal of this subtask is to have a presence less raiden client for now**

As a mediator, Im gonna receive the matrix user id to forward the payment to in the message received. The state machine needs to foward this route information into the next message. The route information *probably* (?) needs to stay complete (All matrix user ids from the route)

**Comment by ezdac** [comment]
I would argue that for a path of nodes `1, ..., i, ...,n`, where 1 is the initiator, i is the current node and n is the target, node i would only have to know user-ids for `1,i-1, i, ...,n` and forward `1, i, ...,n` to the next node.
Easier would be to simply forward the whole path.

**Comment by ezdac** [comment]
It should even be:
As mediator node n, I only receive `1, i+1, ...,n` and send out `1, i+2, ...,n` to user-id `i+1`, 
since the receiver of a message will be informed about the incoming user-id `i-1` by listening for incoming messages? He would then have to memorise user-id `i-1` or the later secret-reveal.

Conceptually
- the initiator creates:
```python
metadata = Metadata(
    forward_user_ids=pfs_user_ids[2:],
    initiator_user_id=my_user_id
)
assert my_user_id == pfs_user_ids[0]
send(metadata, pfs_user_ids[1])
```

The next hop mediates the metadata:
```python
new_metadata = Metadata(
    forward_user_ids=received_metadata.forward_user_ids[1:],
    initiator_user_id=received_metadata.initiator_user_id
)
send(new_metadata, received_metadata.sender_user_id)
```
For this exact approach, we might have to inject the user-id of the sender in to the message upon receiving it from the transport, in the same way we extract it before sending it in the transport.

Should we favor simplicity over some more privacy / smaller message size and just hand over the whole path for each hop?

**Comment by ezdac** [comment]
My vote is probably on simplicity, since we could query the PFS for user-ids anyways (currently also non-neighbours can request user-ids) - anyone voting for keeping the message size as small as possible?

**Comment by fredo** [comment]
At least in the first phase, I definetly suggest simplicity to reach the `durchstich` as fast as possible

**Comment by ezdac** [comment]
Since #6848 is in progress, if you pick this issue up either contact me, or wait for the dependent issue to have an open PR.

**Comment by ezdac** [comment]
Actually, in order to go with sending the whole path, we would have to also modify the way 
we currently send the route-metadata forward (the actual route addresses).
Atm, the metadata is pruned, so that a mediator only receives the route from his address up to the target address.

Keeping this behaviour, but still forwarding the whole path of user-ids feels a little inconsistent and odd to me.
Also, the way the user-ids are constructed one could extract the whole path of addresses from the user-ids anyways.

**Comment by ezdac** [comment]
Ok, after talking with @ulope it seems we agreed before that we forward this as a mapping,
so that all nodes are removed as they would be by pruning the route on each hop, except for the initiator-user-id, which should not be removed.

**Comment by ezdac** [comment]
This was implemented in #6906

A node will include ALL route-metadata for all addresses in the path - this is because the next hop 
needs to know the user-id of the sending node (since we don't want to refactor the transport heavily) and sending all metadata is a more simplistic design for now.
